crates/tx5-go-pion-sys/build.rs

```diff
@@ -56,122 +133,45 @@ fn go_unzip_vendor() {
     .expect("failed to extract vendor zip file");
 }
 
-fn go_build() {
-    let out_dir = std::env::var("OUT_DIR")
-        .map(std::path::PathBuf::from)
-        .expect("error reading out dir");
-
-    let mut lib_path = out_dir.clone();
-
-    let tgt_os = std::env::var("CARGO_CFG_TARGET_OS").unwrap();
-
-    match tgt_os.as_str() {
-        "macos" | "ios" | "tvos" => {
-            lib_path.push("go-pion-webrtc.dylib");
-        }
-        "windows" => {
-            lib_path.push("go-pion-webrtc.dll");
-        }
-        _ => {
-            lib_path.push("go-pion-webrtc.so");
-        }
-    }
-
-    let mut cache = out_dir.clone();
+fn go_build_cmd(
+    out_dir: &std::path::Path,
+    lib_path: &std::path::Path,
+    build_mode: &str,
+) {
+    let mut cache = std::path::PathBuf::from(out_dir);
     cache.push("go-build");
 
-    let manifest_path = std::env::var("CARGO_MANIFEST_DIR")
-        .map(std::path::PathBuf::from)
-        .expect("error reading manifest dir");
-
-    let cp = |f: &'static str| {
-        let mut a = manifest_path.clone();
-        a.push(f);
-        let mut b = out_dir.clone();
-        b.push(f);
-        std::fs::copy(a, b).expect("failed to copy go file");
-    };
-
-    cp("buffer.go");
-    cp("const.go");
-    cp("datachannel.go");
-    cp("main.go");
-    cp("peerconnection.go");
-    cp("go.sum");
-    cp("go.mod");
-
     let mut cmd = Command::new("go");
 
-    // add some cross-compilation translators:
-    match std::env::var("CARGO_CFG_TARGET_ARCH").unwrap().as_str() {
-        "arm" => {
-            cmd.env("GOARCH", "arm");
-        }
-        "aarch64" => {
-            cmd.env("GOARCH", "arm64");
-        }
-        "x86_64" => {
-            cmd.env("GOARCH", "amd64");
-        }
-        "x86" => {
-            cmd.env("GOARCH", "386");
-        }
-        _ => (),
+    if let Some(go_arch) = TARGET.go_arch {
+        cmd.env("GOARCH", go_arch);
     }
 
-    // and for the os
-    match tgt_os.as_str() {
-        "windows" => {
-            cmd.env("GOOS", "windows");
-        }
-        "macos" => {
-            cmd.env("GOOS", "darwin");
-        }
-        "ios" => {
-            cmd.env("GOOS", "ios");
-        }
-        "linux" => {
-            cmd.env("GOOS", "linux");
-        }
-        "android" => {
-            cmd.env("GOOS", "android");
-        }
-        "dragonfly" => {
-            cmd.env("GOOS", "dragonfly");
-        }
-        "freebsd" => {
-            cmd.env("GOOS", "freebsd");
-        }
-        "openbsd" => {
-            cmd.env("GOOS", "openbsd");
-        }
-        "netbsd" => {
-            cmd.env("GOOS", "netbsd");
-        }
-        _ => (),
+    if let Some(go_os) = TARGET.go_os {
+        cmd.env("GOOS", go_os);
```

Comment:
GOOS should always be set right? I wonder if neither the go_os or the go_arch should be optional and instead of the `None` in the `default` impl above this should be a build error for platforms we don't recognise?

Reply:
Technically GOOS only needs to be set if we are cross-compiling. Otherwise it will just default to whatever it detects the host system to be. But yes, for our purposes, I think we want tighter control over this. If someone needs to build for a target that errors this list, they can open an issue and we can sort it out then.

Reply:
Done.